include/sentry.h

```diff
@@ -1440,6 +1440,35 @@ SENTRY_EXPERIMENTAL_API void sentry_span_set_data(
 SENTRY_EXPERIMENTAL_API void sentry_span_remove_data(
     sentry_value_t span, const char *key);
 
+typedef enum {
+    SENTRY_SPAN_STATUS_OK,
+    SENTRY_SPAN_STATUS_CANCELLED,
+    SENTRY_SPAN_STATUS_UNKNOWN,
+    SENTRY_SPAN_STATUS_INVALID_ARGUMENT,
+    SENTRY_SPAN_STATUS_DEADLINE_EXCEEDED,
+    SENTRY_SPAN_STATUS_NOT_FOUND,
+    SENTRY_SPAN_STATUS_ALREADY_EXISTS,
+    SENTRY_SPAN_STATUS_PERMISSION_DENIED,
+    SENTRY_SPAN_STATUS_RESOURCE_EXHAUSTED,
+    SENTRY_SPAN_STATUS_FAILED_PRECONDITION,
+    SENTRY_SPAN_STATUS_ABORTED,
+    SENTRY_SPAN_STATUS_OUT_OF_RANGE,
+    SENTRY_SPAN_STATUS_UNIMPLEMENTED,
+    SENTRY_SPAN_STATUS_INTERNAL_ERROR,
+    SENTRY_SPAN_STATUS_UNAVAILABLE,
+    SENTRY_SPAN_STATUS_DATA_LOSS,
+    SENTRY_SPAN_STATUS_UNAUTHENTICATED
+} sentry_span_status_t;
```

Comment:
i think these enum variants, as well as the num itself could use some amount of documentation (see: https://github.com/getsentry/sentry-native/pull/646/files#diff-56ab308ccaccf29afea8b2668e5d95cf96266f1ca0ef9849863ad601dfac4be3R9-R65).

a link to the documentation in the docstring for the enum itself would also help greatly (https://develop.sentry.dev/sdk/event-payloads/span/).

Reply:
thinking about it a little, i think i'm fine with keeping all of the variants here for the time being instead of hiding them away in `sentry_tracing.h`, so we don't need to use the trick that i'd mentioned earlier to hide this enum's members.

Reply:
I copied over the documentation from your PR. Two questions, though:
1. What is the `UNDEFINED` status used for?
2. Why are `UNDEFINED` and `OK` both 0?